Source/WebCore/page/ElementTargetingController.cpp

```diff
@@ -825,6 +825,38 @@ std::pair<Vector<Ref<Node>>, RefPtr<Element>> ElementTargetingController::findNo
     return { WTFMove(potentialCandidates), WTFMove(foundElement) };
 }
 
+static Vector<Ref<Element>> filterRedundantNearbyTargets(HashSet<Ref<Element>>&& unfilteredNearbyTargets)
+{
+    HashMap<Ref<Element>, bool> shouldKeepCache;
```

Comment:
Nit: Is it not okay to make this a `HashSet<Ref<Element>>`? Where presence indicates the `true` value for the element key, and absence indicates `false`. That would be a little bit more space efficient.

Basically we would condition adding to the set like such:

```
for (auto& ancestor : ancestorsOfTarget) {
    if (shouldKeep)
        shouldKeepCache.add(ancestor);
}
```

Reply:
Hm.. I don't think this would work, since this cache is currently used like so:
• `true` → keep
• `false` → discard
• (missing key) → not cached yet

Reply:
I considered using 2 `HashSet`s instead (one representing keep and another representing discard), but ultimately went with a single map.

Reply:
Okay sounds good, I misunderstood the code!